### DISCUSSION TITLE
Rewrites are in conflict with i18n

### DISCUSSION THREAD

**Discussion by yunsii** [vercel/next.js#68855] [Help]
### Link to the code that reproduces this issue

https://github.com/yunsii/reproduciton-nextjs-rewrites-i18n

### To Reproduce

```ts
{
  source: "/:path((?!api/).*)",
  has: [
    {
      type: "host",
      value: "localhost",
    },
  ],
  destination: `/test.com/:path*`,
}
```

Without i18n config, the rewrites config works very well. But with i18n config, the homepage localhost:3000 goes to 404, I have to add extra rewrite rule:

```ts
{
  source: "/",
  has: [
    {
      type: "host",
      value: "localhost",
    },
  ],
  destination: `/test.com`,
}
```

to make the homepage work as expected.

### Current vs. Expected behavior

I18n config should not affect rewrites config, isn't it?

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: #1 SMP Thu Oct 5 21:02:42 UTC 2023
  Available memory (MB): 40115
  Available CPU cores: 16
Binaries:
  Node: 20.12.2
  npm: 10.5.0
  Yarn: 1.22.15
  pnpm: 6.11.0
Relevant Packages:
  next: 14.2.5 // Latest available version is detected (14.2.5).
  eslint-config-next: 14.2.5
  react: 18.3.1
  react-dom: 18.3.1
  typescript: 5.5.4
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Internationalization (i18n), Navigation, Pages Router

### Which stage(s) are affected? (Select all that apply)

next dev (local)

**Comment by samcx** [comment]
@yunsii Thank you for submitting an issue!

Curious, does it work with `beforeFiles` instead?

**Comment by yunsii** [comment]
> Curious, does it work with `beforeFiles` instead?

All requests return 404 if I change `afterFiles` to `beforeFiles` directly 😦

**Comment by samcx** [comment]
@yunsii Have you taken a look at this in our documentation → https://nextjs.org/docs/pages/api-reference/next-config-js/rewrites#rewrites-with-i18n-support?

**Reply by yunsii** [reply]
I know that, but it seems still no answer for my question, isn't it?
